tests/initialize_holder_rewards.rs

```diff
@@ -0,0 +1,459 @@
+    {
+        context.set_account(
+            &holder_rewards_pool,
+            &AccountSharedData::new_data(100_000_000, &vec![5; 165], &Pubkey::new_unique())
+                .unwrap(),
+        );
+    }
+
+    let instruction =
+        initialize_holder_rewards(&holder_rewards_pool, &holder_rewards, &token_account, &mint);
+
+    let transaction = Transaction::new_signed_with_payer(
+        &[instruction],
+        Some(&context.payer.pubkey()),
+        &[&context.payer],
+        context.last_blockhash,
+    );
+
+    let err = context
+        .banks_client
+        .process_transaction(transaction)
+        .await
+        .unwrap_err()
+        .unwrap();
+
+    assert_eq!(
+        err,
+        TransactionError::InstructionError(0, InstructionError::InvalidAccountOwner)
+    );
+}
+
+#[tokio::test]
+async fn fail_holder_rewards_pool_incorrect_address() {
+    let owner = Pubkey::new_unique();
+    let mint = Pubkey::new_unique();
+
+    let token_account = get_associated_token_address(&owner, &mint);
+    let holder_rewards = get_holder_rewards_address(&token_account);
+    let holder_rewards_pool = Pubkey::new_unique(); // Incorrect holder rewards pool address.
+
+    let mut context = setup().start_with_context().await;
+    setup_holder_rewards_pool_account(&mut context, &holder_rewards_pool, 0).await;
+    setup_token_account(&mut context, &token_account, &owner, &mint, 0).await;
+    setup_mint(&mut context, &mint, &Pubkey::new_unique(), 0).await;
+
+    let instruction =
+        initialize_holder_rewards(&holder_rewards_pool, &holder_rewards, &token_account, &mint);
+
+    let transaction = Transaction::new_signed_with_payer(
+        &[instruction],
+        Some(&context.payer.pubkey()),
+        &[&context.payer],
+        context.last_blockhash,
+    );
+
+    let err = context
+        .banks_client
+        .process_transaction(transaction)
+        .await
+        .unwrap_err()
+        .unwrap();
+
+    assert_eq!(
+        err,
+        TransactionError::InstructionError(
+            0,
+            InstructionError::Custom(PaladinRewardsError::IncorrectHolderRewardsPoolAddress as u32)
+        )
+    );
+}
+
+#[tokio::test]
+async fn fail_holder_rewards_pool_invalid_data() {
+    let owner = Pubkey::new_unique();
+    let mint = Pubkey::new_unique();
+
+    let token_account = get_associated_token_address(&owner, &mint);
+    let holder_rewards = get_holder_rewards_address(&token_account);
+    let holder_rewards_pool = get_holder_rewards_pool_address(&mint);
+
+    let mut context = setup().start_with_context().await;
+    setup_token_account(&mut context, &token_account, &owner, &mint, 0).await;
+    setup_mint(&mut context, &mint, &Pubkey::new_unique(), 0).await;
+
+    // Set up a holder rewards pool account with invalid data.
+    {
+        context.set_account(
+            &holder_rewards_pool,
+            &AccountSharedData::new_data(100_000_000, &vec![5; 165], &spl_token_2022::id())
```

Comment:
This test looks very similar to `fail_holder_rewards_pool_incorrect_owner` -- maybe the owner should be paladin_rewards_program::id() instead of spl_token_2022::id() to test the failing deserialization condition?

Reply:
Yep I think this was a copy-pasta error on my part. Good spot!